--- client/klive.tac.orig	2010-01-07 13:50:59.000000000 +0100
+++ client/klive.tac	2010-01-07 14:29:29.000000000 +0100
@@ -77,7 +77,7 @@
 
 PORT = 4921
 
-import sys, os, struct, md5, re, glob, stat, zlib, socket
+import sys, os, struct, hashlib, re, glob, stat, zlib, socket
 
 from twisted.internet.protocol import DatagramProtocol
 from twisted.internet import reactor
@@ -122,7 +122,7 @@
 	return struct.pack('!H', len(s)) + s
 
 def gen_random_hash():
-	return md5.new(file('/dev/urandom').read(100) + os.uname()[1]).digest()	
+	return hashlib.md5(file('/dev/urandom').read(100) + os.uname()[1]).digest()	
 
 def hexdigest(digest):
 	import string
