diff -Nur supertux-0.3.1.orig/src/lisp/lisp.hpp supertux-0.3.1/src/lisp/lisp.hpp
--- supertux-0.3.1.orig/src/lisp/lisp.hpp	2008-01-03 21:59:26.000000000 +0100
+++ supertux-0.3.1/src/lisp/lisp.hpp	2009-06-02 21:26:17.952497076 +0200
@@ -23,6 +23,7 @@
 #include <string>
 #include <vector>
 #include <assert.h>
+#include <cstdio>
 
 namespace lisp
 {
diff -Nur supertux-0.3.1.orig/src/random_generator.cpp supertux-0.3.1/src/random_generator.cpp
--- supertux-0.3.1.orig/src/random_generator.cpp	2008-01-03 21:59:27.000000000 +0100
+++ supertux-0.3.1/src/random_generator.cpp	2009-06-02 17:56:22.920497466 +0200
@@ -40,6 +40,7 @@
 #include <time.h>
 #include <cassert>
 #include "random_generator.hpp"
+#include <cstdio>
 
 RandomGenerator systemRandom;               // global random number generator
 
