diff -upr supertux-0.3.1-orig/src/addon_manager.cpp supertux-0.3.1/src/addon_manager.cpp
--- supertux-0.3.1-orig/src/addon_manager.cpp	2008-01-04 00:59:28.000000000 +0400
+++ supertux-0.3.1/src/addon_manager.cpp	2008-12-09 20:51:24.328040904 +0400
@@ -25,6 +25,7 @@
 #include <physfs.h>
 #include <sys/stat.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include "addon_manager.hpp"
 #include "config.h"
 #include "log.hpp"
diff -upr supertux-0.3.1-orig/src/console.hpp supertux-0.3.1/src/console.hpp
--- supertux-0.3.1-orig/src/console.hpp	2008-01-04 00:59:27.000000000 +0400
+++ supertux-0.3.1/src/console.hpp	2008-12-09 20:51:24.328040904 +0400
@@ -22,6 +22,7 @@
 
 #include <list>
 #include <map>
+#include <memory>
 #include <vector>
 #include <string>
 #include <sstream>
diff -upr supertux-0.3.1-orig/src/lisp/lexer.cpp supertux-0.3.1/src/lisp/lexer.cpp
--- supertux-0.3.1-orig/src/lisp/lexer.cpp	2008-01-04 00:59:26.000000000 +0400
+++ supertux-0.3.1/src/lisp/lexer.cpp	2008-12-09 20:51:24.328040904 +0400
@@ -21,6 +21,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <iostream>
+#include <string.h>
 
 #include "lexer.hpp"
 
diff -upr supertux-0.3.1-orig/src/main.cpp supertux-0.3.1/src/main.cpp
--- supertux-0.3.1-orig/src/main.cpp	2008-01-04 00:59:27.000000000 +0400
+++ supertux-0.3.1/src/main.cpp	2008-12-09 20:51:24.328040904 +0400
@@ -431,9 +431,9 @@ void init_video()
 
   // set icon
   #ifdef MACOSX
-  char* icon_fname = "images/engine/icons/supertux-256x256.png";
+  const char *icon_fname = "images/engine/icons/supertux-256x256.png";
   #else
-  char* icon_fname = "images/engine/icons/supertux.xpm";
+  const char *icon_fname = "images/engine/icons/supertux.xpm";
   #endif
   SDL_Surface* icon = IMG_Load_RW(get_physfs_SDLRWops(icon_fname), true);
   if(icon != 0) {
diff -upr supertux-0.3.1-orig/src/sprite/sprite_manager.hpp supertux-0.3.1/src/sprite/sprite_manager.hpp
--- supertux-0.3.1-orig/src/sprite/sprite_manager.hpp	2008-01-04 00:59:26.000000000 +0400
+++ supertux-0.3.1/src/sprite/sprite_manager.hpp	2008-12-09 20:51:24.328040904 +0400
@@ -21,6 +21,7 @@
 #define SUPERTUX_SPRITE_MANAGER_H
 
 #include <map>
+#include <string>
 
 class SpriteData;
 class Sprite;
diff -upr supertux-0.3.1-orig/src/squirrel/include/squirrel.h supertux-0.3.1/src/squirrel/include/squirrel.h
--- supertux-0.3.1-orig/src/squirrel/include/squirrel.h	2008-01-04 00:59:25.000000000 +0400
+++ supertux-0.3.1/src/squirrel/include/squirrel.h	2008-12-09 20:49:46.180540970 +0400
@@ -42,16 +42,19 @@ extern "C" {
 #ifdef _SQ64
 #ifdef _MSC_VER
 typedef __int64 SQInteger;
+#define _SC_INT_FMT "%Id"
 typedef unsigned __int64 SQUnsignedInteger;
 typedef unsigned __int64 SQHash; /*should be the same size of a pointer*/
 #else
 typedef long SQInteger;
+#define _SC_INT_FMT "%ld"
 typedef unsigned long SQUnsignedInteger;
 typedef unsigned long SQHash; /*should be the same size of a pointer*/
 #endif
 typedef int SQInt32; 
 #else 
 typedef int SQInteger;
+#define _SC_INT_FMT "%d"
 typedef int SQInt32; /*must be 32 bits(also on 64bits processors)*/
 typedef unsigned int SQUnsignedInteger;
 typedef unsigned int SQHash; /*should be the same size of a pointer*/
diff -upr supertux-0.3.1-orig/src/squirrel/sqstdlib/sqstdrex.cpp supertux-0.3.1/src/squirrel/sqstdlib/sqstdrex.cpp
--- supertux-0.3.1-orig/src/squirrel/sqstdlib/sqstdrex.cpp	2008-01-04 00:59:25.000000000 +0400
+++ supertux-0.3.1/src/squirrel/sqstdlib/sqstdrex.cpp	2008-12-09 20:49:46.180540970 +0400
@@ -483,7 +483,7 @@ static const SQChar *sqstd_rex_matchnode
 			return cur;
 	}				 
 	case OP_WB:
-		if(str == exp->_bol && !isspace(*str)
+		if((str == exp->_bol && !isspace(*str))
 		 || (str == exp->_eol && !isspace(*(str-1)))
 		 || (!isspace(*str) && isspace(*(str+1)))
 		 || (isspace(*str) && !isspace(*(str+1))) ) {
diff -upr supertux-0.3.1-orig/src/squirrel/sqstdlib/sqstdstream.cpp supertux-0.3.1/src/squirrel/sqstdlib/sqstdstream.cpp
--- supertux-0.3.1-orig/src/squirrel/sqstdlib/sqstdstream.cpp	2008-01-04 00:59:25.000000000 +0400
+++ supertux-0.3.1/src/squirrel/sqstdlib/sqstdstream.cpp	2008-12-09 20:49:46.180540970 +0400
@@ -277,7 +277,7 @@ void init_streamclass(HSQUIRRELVM v)
 	sq_pop(v,1);
 }
 
-SQRESULT declare_stream(HSQUIRRELVM v,SQChar* name,SQUserPointer typetag,SQChar* reg_name,SQRegFunction *methods,SQRegFunction *globals)
+SQRESULT declare_stream(HSQUIRRELVM v,const SQChar* name,SQUserPointer typetag,const SQChar* reg_name,SQRegFunction *methods,SQRegFunction *globals)
 {
 	if(sq_gettype(v,-1) != OT_TABLE)
 		return sq_throwerror(v,_SC("table expected"));
diff -upr supertux-0.3.1-orig/src/squirrel/sqstdlib/sqstdstream.h supertux-0.3.1/src/squirrel/sqstdlib/sqstdstream.h
--- supertux-0.3.1-orig/src/squirrel/sqstdlib/sqstdstream.h	2008-01-04 00:59:25.000000000 +0400
+++ supertux-0.3.1/src/squirrel/sqstdlib/sqstdstream.h	2008-12-09 20:49:46.180540970 +0400
@@ -14,5 +14,5 @@ SQInteger _stream_eos(HSQUIRRELVM v);
 SQInteger _stream_flush(HSQUIRRELVM v);
 
 #define _DECL_STREAM_FUNC(name,nparams,typecheck) {_SC(#name),_stream_##name,nparams,typecheck}
-SQRESULT declare_stream(HSQUIRRELVM v,SQChar* name,SQUserPointer typetag,SQChar* reg_name,SQRegFunction *methods,SQRegFunction *globals);
+SQRESULT declare_stream(HSQUIRRELVM v,const SQChar* name,SQUserPointer typetag,const SQChar* reg_name,SQRegFunction *methods,SQRegFunction *globals);
 #endif /*_SQSTD_STREAM_H_*/
diff -upr supertux-0.3.1-orig/src/squirrel/squirrel/sqapi.cpp supertux-0.3.1/src/squirrel/squirrel/sqapi.cpp
--- supertux-0.3.1-orig/src/squirrel/squirrel/sqapi.cpp	2008-01-04 00:59:26.000000000 +0400
+++ supertux-0.3.1/src/squirrel/squirrel/sqapi.cpp	2008-12-09 20:49:46.180540970 +0400
@@ -774,7 +774,7 @@ SQRESULT sq_rawdeleteslot(HSQUIRRELVM v,
 		_table(*self)->Remove(key);
 	}
 	if(pushval != 0)
-		if(pushval)	v->GetUp(-1) = t;
+		v->GetUp(-1) = t;
 	else
 		v->Pop(1);
 	return SQ_OK;
Only in supertux-0.3.1/src/squirrel/squirrel: sqapi.cpp.orig
diff -upr supertux-0.3.1-orig/src/squirrel/squirrel/sqcompiler.cpp supertux-0.3.1/src/squirrel/squirrel/sqcompiler.cpp
--- supertux-0.3.1-orig/src/squirrel/squirrel/sqcompiler.cpp	2008-01-04 00:59:26.000000000 +0400
+++ supertux-0.3.1/src/squirrel/squirrel/sqcompiler.cpp	2008-12-09 20:49:46.180540970 +0400
@@ -778,7 +778,7 @@ public:
 			SQInteger val = _fs->PopTarget();
 			SQInteger key = _fs->PopTarget();
 			SQInteger attrs = hasattrs ? _fs->PopTarget():-1;
-			assert(hasattrs && attrs == key-1 || !hasattrs);
+			assert((hasattrs && attrs == key-1) || !hasattrs);
 			unsigned char flags = (hasattrs?NEW_SLOT_ATTRIBUTES_FLAG:0)|(isstatic?NEW_SLOT_STATIC_FLAG:0);
 			SQInteger table = _fs->TopTarget(); //<<BECAUSE OF THIS NO COMMON EMIT FUNC IS POSSIBLE
 			_fs->AddInstruction(_OP_NEWSLOTA, flags, table, key, val);
Only in supertux-0.3.1/src/squirrel/squirrel: sqcompiler.cpp.orig
diff -upr supertux-0.3.1-orig/src/squirrel/squirrel/sqdebug.cpp supertux-0.3.1/src/squirrel/squirrel/sqdebug.cpp
--- supertux-0.3.1-orig/src/squirrel/squirrel/sqdebug.cpp	2008-01-04 00:59:26.000000000 +0400
+++ supertux-0.3.1/src/squirrel/squirrel/sqdebug.cpp	2008-12-09 20:49:46.180540970 +0400
@@ -57,7 +57,7 @@ SQString *SQVM::PrintObjVal(const SQObje
 	switch(type(o)) {
 	case OT_STRING: return _string(o);
 	case OT_INTEGER:
-		scsprintf(_sp(rsl(NUMBER_MAX_CHAR+1)), _SC("%d"), _integer(o));
+		scsprintf(_sp(rsl(NUMBER_MAX_CHAR+1)), _SC(_SC_INT_FMT), _integer(o));
 		return SQString::Create(_ss(this), _spval);
 		break;
 	case OT_FLOAT:
diff -upr supertux-0.3.1-orig/src/squirrel/squirrel/sqfuncstate.cpp supertux-0.3.1/src/squirrel/squirrel/sqfuncstate.cpp
--- supertux-0.3.1-orig/src/squirrel/squirrel/sqfuncstate.cpp	2008-01-04 00:59:25.000000000 +0400
+++ supertux-0.3.1/src/squirrel/squirrel/sqfuncstate.cpp	2008-12-09 20:50:37.268040640 +0400
@@ -79,7 +79,7 @@ void DumpLiteral(SQObjectPtr &o)
 	switch(type(o)){
 		case OT_STRING:	scprintf(_SC("\"%s\""),_stringval(o));break;
 		case OT_FLOAT: scprintf(_SC("{%f}"),_float(o));break;
-		case OT_INTEGER: scprintf(_SC("{%d}"),_integer(o));break;
+		case OT_INTEGER: scprintf(_SC("{" _SC_INT_FMT "}"),_integer(o));break;
 		default: assert(0); break; //shut up compiler
 	}
 }
Only in supertux-0.3.1/src/squirrel/squirrel: sqfuncstate.cpp.orig
Only in supertux-0.3.1/src/squirrel/squirrel: sqfuncstate.cpp.rej
diff -upr supertux-0.3.1-orig/src/squirrel/squirrel/sqvm.cpp supertux-0.3.1/src/squirrel/squirrel/sqvm.cpp
--- supertux-0.3.1-orig/src/squirrel/squirrel/sqvm.cpp	2008-01-04 00:59:26.000000000 +0400
+++ supertux-0.3.1/src/squirrel/squirrel/sqvm.cpp	2008-12-09 20:49:46.180540970 +0400
@@ -231,7 +231,7 @@ void SQVM::ToString(const SQObjectPtr &o
 		scsprintf(_sp(rsl(NUMBER_MAX_CHAR+1)),_SC("%g"),_float(o));
 		break;
 	case OT_INTEGER:
-		scsprintf(_sp(rsl(NUMBER_MAX_CHAR+1)),_SC("%d"),_integer(o));
+		scsprintf(_sp(rsl(NUMBER_MAX_CHAR+1)),_SC(_SC_INT_FMT),_integer(o));
 		break;
 	case OT_BOOL:
 		scsprintf(_sp(rsl(6)),_integer(o)?_SC("true"):_SC("false"));
@@ -645,7 +645,7 @@ bool SQVM::IsEqual(SQObjectPtr &o1,SQObj
 
 bool SQVM::IsFalse(SQObjectPtr &o)
 {
-	if((type(o) & SQOBJECT_CANBEFALSE) && ( (type(o) == OT_FLOAT) && (_float(o) == SQFloat(0.0)) )
+	if(((type(o) & SQOBJECT_CANBEFALSE) && ( (type(o) == OT_FLOAT) && (_float(o) == SQFloat(0.0)) ))
 		|| (_integer(o) == 0) ) { //OT_NULL|OT_INTEGER|OT_BOOL
 		return true;
 	}
Only in supertux-0.3.1/src/squirrel/squirrel: sqvm.cpp.orig
diff -upr supertux-0.3.1-orig/src/textscroller.hpp supertux-0.3.1/src/textscroller.hpp
--- supertux-0.3.1-orig/src/textscroller.hpp	2008-01-04 00:59:27.000000000 +0400
+++ supertux-0.3.1/src/textscroller.hpp	2008-12-09 20:51:24.328040904 +0400
@@ -24,6 +24,7 @@
 #include <vector>
 #include <string>
 #include <map>
+#include <memory>
 
 #include "screen.hpp"
 #include "math/vector.hpp"
diff -upr supertux-0.3.1-orig/src/title.cpp supertux-0.3.1/src/title.cpp
--- supertux-0.3.1-orig/src/title.cpp	2008-01-04 00:59:28.000000000 +0400
+++ supertux-0.3.1/src/title.cpp	2008-12-09 20:51:24.328040904 +0400
@@ -32,6 +32,7 @@
 #include <SDL.h>
 #include <SDL_image.h>
 #include <physfs.h>
+#include <algorithm>
 
 #include "title.hpp"
 #include "mainloop.hpp"
diff -upr supertux-0.3.1-orig/src/video/sdl_texture.hpp supertux-0.3.1/src/video/sdl_texture.hpp
--- supertux-0.3.1-orig/src/video/sdl_texture.hpp	2008-01-04 00:59:22.000000000 +0400
+++ supertux-0.3.1/src/video/sdl_texture.hpp	2008-12-09 20:51:24.328040904 +0400
@@ -21,6 +21,7 @@
 #define __SDL_TEXTURE_HPP__
 
 #include <config.h>
+#include <algorithm>
 
 #include <SDL.h>
 
