diff --git a/src/callbacks.c b/src/callbacks.c
index f5a4bd4..44fb7f6 100644
--- a/src/callbacks.c
+++ b/src/callbacks.c
@@ -160,7 +160,8 @@ on_modes_combo_changed                 (GtkComboBox     *combobox,
 	int mode_id;
 	
 	model = gtk_combo_box_get_model (combobox);
-	gtk_combo_box_get_active_iter (combobox, &iter);
+	if (!gtk_combo_box_get_active_iter (combobox, &iter))
+		return;
   	gtk_tree_model_get (model, &iter,
 		      COL_MODE_ID, &mode_id,
 		      -1);
