--- plugins/xrandr/gsd-xrandr-manager.c.orig	2009-04-22 15:04:15.033062130 +0200
+++ plugins/xrandr/gsd-xrandr-manager.c	2009-04-22 15:08:24.849292358 +0200
@@ -118,7 +118,7 @@
             ev->type == ClientMessage &&
             ev->xclient.message_type == gnome_randr_xatom ()) {
 
-                gnome_rr_config_apply_stored (screen);
+                gnome_rr_config_apply_stored (screen, NULL);
 
                 return GDK_FILTER_REMOVE;
         }
@@ -146,7 +146,7 @@
                 /* FIXME: here we should cycle between valid
                  * configurations, and save them
                  */
-                gnome_rr_config_apply_stored (manager->priv->rw_screen);
+                gnome_rr_config_apply_stored (manager->priv->rw_screen, NULL);
 
                 return GDK_FILTER_CONTINUE;
         }
@@ -391,7 +391,7 @@
 
                 output->rotation = rotation_to_test;
 
-                if (gnome_rr_config_applicable (priv->configuration, priv->rw_screen)) {
+                if (gnome_rr_config_applicable (priv->configuration, priv->rw_screen, NULL)) {
                         (*out_num_rotations)++;
                         (*out_rotations) |= rotation_to_test;
                 }
@@ -456,7 +456,7 @@
 
         error = NULL;
         if (gnome_rr_config_save (priv->configuration, &error)) {
-                if (!gnome_rr_config_apply_stored (priv->rw_screen)) {
+                if (!gnome_rr_config_apply_stored (priv->rw_screen, NULL)) {
                         error_dialog (_("The selected rotation could not be applied"),
                                       _("An error occurred while configuring the screen"));
                         /* FIXME: that message is really useless.  Make
@@ -694,7 +694,7 @@
         g_debug ("Starting xrandr manager");
 
         manager->priv->rw_screen = gnome_rr_screen_new (
-                gdk_screen_get_default (), on_randr_event, manager);
+                gdk_screen_get_default (), on_randr_event, manager, NULL);
 
         if (manager->priv->rw_screen == NULL) {
                 g_set_error (error, 0, 0, "Failed to initialize XRandR extension");
@@ -728,7 +728,7 @@
                 gdk_error_trap_pop ();
         }
 
-        gnome_rr_config_apply_stored (manager->priv->rw_screen);
+        gnome_rr_config_apply_stored (manager->priv->rw_screen, NULL);
 
         gdk_window_add_filter (gdk_get_default_root_window(),
                                (GdkFilterFunc)event_filter,
