--- git-1.7.6.orig/git-svn.perl	2011-06-27 00:26:13.000000000 +0200
+++ git-1.7.6/git-svn.perl	2011-07-07 14:28:45.000000000 +0200
@@ -4930,6 +4930,9 @@
 
 sub _auth_providers () {
 	[
+	  $SVN::Core::VERSION lt '1.6' ? () :
+	    @{SVN::Core::auth_get_platform_specific_client_providers(
+	      undef,undef)}
 	  SVN::Client::get_simple_provider(),
 	  SVN::Client::get_ssl_server_trust_file_provider(),
 	  SVN::Client::get_simple_prompt_provider(
